tests/scripts/test_integration.py

```diff
@@ -19,6 +21,22 @@ def test_eval_subcommand(
     main(args)
 
 
+def test_eval_subcommand_fails_when_not_enough_data_given(
+    text_dataset_path: Path, gpt2_tiny_random_model_local_path: Path, tmp_path: Path
+):
+    args = (
+        f"--log_level DEBUG eval --data.name {text_dataset_path}"
+        f" --model.name {gpt2_tiny_random_model_local_path}"
+        " --record_logit_stats"
+        " --tokens 100000 --max_length 128"
```

Comment:
Is `max_length` the length of the data set? Or does this rely on the length of `text_dataset_path`? I'd appreciate a clarification.

Reply:
That's the maximum number of tokens in the batch. I guess that is a slightly confusing name.

Reply:
I still don't understand when the error is supposed to be triggered. It's an error about insufficient *total* trainign data, right? Not data per batch? So why is the max tokens per batch relevant?

Reply:
I guess for this test it isn't. I'm testing on a very small dev set. If you don't pass this flag you run into other errors since there is not enough data to even form a single batch using the default batch size.